### PR TITLE
make SuiExecuteTransactionResponse struct but not enum

### PR DESCRIPTION
**Pull request by longbowlu**
as title, this is to clean up the leftover  from the Quorum Driver refactor

### CODE REVIEW

File: crates/sui-json-rpc-types/src/lib.rs [hunk 1]
```diff
@@ -400,15 +400,12 @@ impl Display for SuiParsedTransactionResponse {
...
-        // before this response.
-        confirmed_local_execution: bool,
-    },
+pub struct SuiExecuteTransactionResponse {
+    pub certificate: SuiCertifiedTransaction,
+    pub effects: SuiCertifiedTransactionEffects,
+    // If the transaction is confirmed to be executed locally
+    // before this response.
+    pub confirmed_local_execution: bool,
 }
```
Comment (666lcz) [inline crates/sui-json-rpc-types/src/lib.rs:409]: To make the TS-SDK backward compatible with this change: we would need to mark the [existing TS type](https://github.com/MystenLabs/sui/blob/main/sdk/typescript/src/types/transactions.ts#L229-L241) as deprecated and add a new one, and return the right type in https://github.com/MystenLabs/sui/blob/main/sdk/typescript/src/providers/json-rpc-provider.ts#L578-L609 based on the [RPC version](https://github.com/MystenLabs/sui/blob/8474242afcab175830b64817821a37a497edb3a1/sdk/typescript/src/providers/json-rpc-provider.ts#L147)

**Review by 666lcz — CHANGES_REQUESTED** [review]
Thanks for cleaning this up! We would need to make the corresponding TS-SDK change before merging this

File: crates/sui-json-rpc-types/src/lib.rs (hunk 1 above)
Comment (longbowlu) [reply crates/sui-json-rpc-types/src/lib.rs:409]: thanks, will follow the instructions to make the changes

**Comment by longbowlu** [comment]
> Just want to point out that this may become an enum again, latter. Right now there is one time of finalized transaction, which is a certified effects. But latter down the road, we may want to return effects along with epoch/checkpoint number to indicate that this tx was finalized at a specific epoch/checkpoint. What do you think?


good point. Do you prefer to leave this as-is or convert it as an enum in the future (another breaking change)? I have no strong opinions but if we plan to do that relatively soon (e.g. before wave3) then we probably can close this PR

**Comment by lxfind** [comment]
Sounds like it will still be a struct, with potentially an enum proof in it. So this change is on the right direction.

**Comment by lxfind** [comment]
Can we not make SDK backward compatible right now? There are quite a few changes coming to this interface, and I think it would be a huge burden to try to keep it backward compatible.
Specifically, we will need to remove the Certificate from the response, and also make the effects raw effects instead of effects cert

File: sdk/typescript/src/types/transactions.ts [hunk 2]
```diff
@@ -237,6 +237,11 @@ export const SuiExecuteTransactionResponse = union([
       confirmed_local_execution: boolean(),
     }),
   }),
+  object({
+    certificate: CertifiedTransaction,
+    effects: SuiCertifiedTransactionEffects,
+    confirmed_local_execution: boolean(),
+  }),
```
Comment (longbowlu) [inline sdk/typescript/src/types/transactions.ts:244]: @666lcz  I didn't mark this one as deprecated but I expand the enum instead. All sdk e2e tests pass

File: crates/sui-json-rpc-types/src/lib.rs (hunk 1 above)
Comment (longbowlu) [reply crates/sui-json-rpc-types/src/lib.rs:421]: https://github.com/MystenLabs/sui/pull/7468/files#r1091322788

File: sdk/typescript/src/types/transactions.ts (hunk 2 above)
Comment (666lcz) [reply sdk/typescript/src/types/transactions.ts:244]: Nice! I think you can remove line 232 (which has been removed from the RPC side long ago). Could you also add a `// TODO: remove after devnet 0.25.0(or 0.24.0?) is released` before line 233? Thanks!

**Review by 666lcz — APPROVED** [review]

**Comment by 666lcz** [comment]
Before merging, could you
1. run `pnpm changeset` in the root directory?
2. double check if the wallet is working properly(I am a bit surprised that I did not see wallet related changes here)

**Comment by longbowlu** [comment]
@666lcz  wallet works

**Pull request merged**